samples/apps/programmability/programmability.cpp

```diff
@@ -240,6 +257,39 @@ namespace programmabilityapp
         auto audit_info = ctx.tx.template rw<AuditInfoValue>(
           fmt::format("{}.audit.info", CUSTOM_ENDPOINTS_NAMESPACE));
         audit_info->put({format, AuditInputContent::BUNDLE, user_id.value()});
+        if (format == AuditInputFormat::COSE)
```

Comment:
This (plus the lines above) seems like something we could, and should compress quite a bit. This requires two main API extensions I think:

1. `AuditInputContent` and `AuditInputFormat` equivalents in `registry.h`, and probably moving `get_body()` there.
2. Either a string or enum to expose error details during replay protection.

This can then be replaced by one, or two calls (I think two is best for clarity, naming the replay protection check is already quite difficult).

There is not a lot of short-term benefit (there are probably not going to be lots more auditable calls), but will be helpful if we look to expose audit getters for example.